### PR TITLE
SceneView: ImageOverlays

### PR DESCRIPTION
**Pull request by hud10837**
Adds `imageOverlays` param and associated `ImageOverlaysCollection` to SceneView.

<details>
<summary>This code snippet will help with ad-hoc testing</summary>

```kotlin
 val imageOverlayCollection = rememberImageOverlayCollection()
    val point = Point(22.321736, 44.697703, SpatialReference.wgs84())
    val buffered = GeometryEngine.bufferOrNull(point, 30.0)
    val envelope = buffered?.extent
    val imageFrame = ImageFrame("https://upload.wikimedia.org/wikipedia/commons/thumb/e/e0/SNice.svg/1200px-SNice.svg.png", envelope!!)
    val imageOverlay = ImageOverlay(imageFrame)
    imageOverlayCollection.add(imageOverlay)
```

</details>

### CODE REVIEW

File: toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/ImageOverlayCollection.kt [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+ *
+ * @since 200.4.0
+ */
+@Composable
+internal fun ImageOverlaysUpdater(
+    imageOverlayCollection: ImageOverlayCollection,
+    sceneView: SceneView
+) {
+    LaunchedEffect(imageOverlayCollection) {
+        // sync up the GeoView with the new graphics overlays
```
Comment (puneet-pdx) [inline toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/ImageOverlayCollection.kt:125]:
```suggestion
        // sync up the GeoView with the new image overlays
```

**Review by puneet-pdx — COMMENTED** [review]
@hud10837 some doc comments. looks good overall. 👍🏽

File: toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/SceneView.kt [hunk 2]
```diff
@@ -204,6 +207,42 @@ private fun ViewpointUpdater(
...
+ *
+ * @since 200.4.0
+ */
+@Composable
+internal fun ImageOverlaysUpdater(
+    imageOverlayCollection: ImageOverlayCollection,
+    sceneView: SceneView
+) {
+    LaunchedEffect(imageOverlayCollection) {
+        // sync up the GeoView with the new image overlays
```
Comment (puneet-pdx) [inline toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/SceneView.kt:222]:
```suggestion
        // sync up the SceneView with the new image overlays
```

**Review by puneet-pdx — APPROVED** [review]

File: toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/ImageOverlayCollection.kt [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+ *
+ */
+
+package com.arcgismaps.toolkit.geocompose
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.Stable
+import com.arcgismaps.mapping.view.ImageOverlay
+import com.arcgismaps.mapping.view.SceneView
```
Comment (changanxian) [inline toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/ImageOverlayCollection.kt:24]: Clean up the unused import

File: toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/ImageOverlayCollection.kt [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        _changed.tryEmit(ChangedEvent.Cleared)
+    }
+
+    /**
+     * Sealed class used to notify the compose SceneView to update the ImageOverlays on the
+     * type of [ChangedEvent].
+     *
+     * @since 200.4.0
+     */
+    internal sealed class ChangedEvent() {
```
Comment (changanxian) [inline toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/ImageOverlayCollection.kt:104]: remove the `()`

File: toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/ImageOverlayCollection.kt [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        class Added(val element: ImageOverlay) : ChangedEvent()
+        class Removed(val element: ImageOverlay) : ChangedEvent()
+        object Cleared : ChangedEvent()
```
Comment (changanxian) [inline toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/ImageOverlayCollection.kt:107]: IDE suggests changing to `data object Cleared`

Comment (hud10837) [reply toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/ImageOverlayCollection.kt]: FYI - I made this change in this PR also for AnalysisOverlayCollection and GraphicsOverlayCollection

**Review by changanxian — APPROVED** [review]

**Pull request merged**